applications/vanilla/models/class.categorymodel.php

```diff
@@ -2527,18 +2527,18 @@ protected static function _MakeTreeChildren($category, $categories, $depthAdj =
      * @since 2.2
      */
     public static function permissionCategory($category) {
-        if (empty($category)) {
+        if (empty($category) || empty($category['PermissionCategoryID'])) {
```

Comment:
A couple lines down, it looks like this function is supposed to support arrays, objects and even integers. Won't this break that support?